File: src/lc_209_min_size_subarray.py
class MinSizeSubarraySum():
    
    def __init__(self, nums, target):
        self.nums   = nums
        self.target = target

    def solve(self):

        #Sliding Windows

        nums    = self.nums
        target  = self.target

        i = j = 0
        sum             = nums[j]
        MAX_ARRAY_SIZE  = 10**5 
        output          = MAX_ARRAY_SIZE + 1 

        print(f"Output = {output}")

        while ( j < len(nums) ):

            if (sum >= target):
                output = min( output, (j - i + 1))

                if (i == j) :
                    i += 1
                    j += 1
                    if ( j < len(nums) ):
                        sum = nums[j]
                else:
                    sum -= nums[i]
                    i += 1
            else:
                j   += 1
                if ( j < len(nums) ):
                    sum += nums[j]

        if (output == MAX_ARRAY_SIZE + 1):
            return 0
        else:
            return output

File: src/lc_209_min_size_subarray.c
#include <stdio.h>

#define MAX_SIZE_ARRAY 100000
#define MIN(a,b) ( (a < b) ? a : b )

int minSubArrayLen(int target, int* nums, int numsSize) 
{
    int output  = MAX_SIZE_ARRAY + 1;

    //Initialize
    int i = 0;
    int j = 0;
    int sum = nums[j];

    printf("Nums size array = %d\n", numsSize);

    while (j < numsSize)
    {
        if ( sum >= target)
        {
            output = MIN(output, (j - i + 1));

            if ( i == j)
            {
                i++;
                j++;
                if (j < numsSize)
                {
                    sum = nums[j];
                }
            }
            else
            {
                sum = sum - nums[i];
                i++;
            }
        }
        else
        {
            j++;
            if (j < numsSize)
            {
                sum = sum + nums[j];
            }            
        }
    }

    if (output == MAX_SIZE_ARRAY + 1)
    {
        return 0;
    }
    else
    {
        return output;
    }    
    
}
